miden-lib/asm/miden/contracts/wallets/basic.masm

```diff
@@ -43,7 +43,6 @@ export.send_asset.1
     push.0 swap loc_store.0 padw push.0.0.0 swapdw loc_load.0
     # => [ASSET, tag, RECIPIENT, ZERO, ZERO, ...]
 
-
-    exec.tx::create_note
+    exec.tx::create_public_note
```

Comment:
This is a bit problematic as now all notes created via `send_asset` will be public notes, and we might need to add another procedure for sending off-chain notes.

So, I wonder if we should do it a bit differently: pass output note type as another parameter on the stack (in addition to `ASSET`, `tag`, and `recipient`). This type can then be used in an - `if-else` statement to determine which procedure for creating notes to all. This way, the caller of the `send_asset` procedure will be able to determine the type of the output note.

Reply:
If we are going to pass note type into procedures like `send_asset`, I wonder if we should even have two separate procedures for creating notes. Maybe it should still be `create_note` - but now it would also take `note_type` as a parameter and perform all required validation.

Reply:
note type is now an argument

Reply:
I have added note_type to send_asset too